src/components/Checkbox/Checkbox.st.css

```diff
@@ -95,9 +118,49 @@
     color: value(CurrentTextColor);
 }
 
+.label.suffixed {
+    flex: 1;
+    white-space: nowrap;
+    display: flex;
+}
+
+.label.suffixed::after {
+    content: '';
+    width: 5px;
+}
+
+.suffix {
+    flex: 0;
+}
+
 /* STATES */
+.root:box:disabled {
+    cursor: default;
+}
+
+.root:box:hover, .root:box:focus-within {
+    border-color: applyOpacity(value(CurrentBoxBorderColor), 1);
+}
+
+.root:box.disabled:hover:not(:checked), .root:box:disabled:focus-within:not(:checked) {
+    border-color: applyOpacity(value(CurrentBoxBorderColor), 0.6);
+}
+
+.root:box:checked {
+    background-color: applyOpacity(value(CurrentIconColor), 0.1);
+    border-color: value(CurrentIconColor);
+}
+
+.root:box:checked:not(:disabled):hover, .root:box:checked:not(:disabled):focus-within {
+    border-color: applyOpacity(value(CurrentIconColor), 0.6);
+}
+
+.root:box:not(:disabled):hover .icon {
+    border-color: applyOpacity(value(CurrentBoxBorderColor), 1);
+}
+
 .root:hover .icon {
-    border-color: applyOpacity(value(CurrentBorderColor), 1);
+    border-color: applyOpacity(value(CurrentBoxBorderColor), 1);
```

Comment:
Shouldn't this stay `CurrentBorderColor`?
It's not under the `box` state

Reply:
Right, I changed it back.